File: C#/0007. Reverse Integer.cs
public class Solution {
    public int Reverse(int x) {
        int flag=1;
        if(x<0){
            flag=-1;
            x=-x;
        }
        
        long y=0;
        while (x>0){
            int tmp=x%10;
            y=y*10+tmp;
            x=x/10;
        }
        
        if(flag==-1){
            y=-y;
        }
        
        if(y>int.MaxValue||y<int.MinValue ){
            return 0;
        }
        return (int)y;
    }
}

File: 20200215Python-China/0007. Reverse Integer.py
class Solution:
    def reverse(self, x: int) -> int:
        flag=True
        if x<0:
            flag=False
            x=-x

        rep=0
        while x>0:
            rep=rep*10+x%10
            x=(x-x%10)//10

        if flag==False:
            rep=-rep

        if rep>2**31-1 or rep<-2**31:
            return 0
        return rep
